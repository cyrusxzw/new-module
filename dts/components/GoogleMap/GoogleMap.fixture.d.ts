export namespace GoogleMapFixture {
  const initialZoom: number;
  namespace center {
    const lat: number;
    const lng: number;
  }
  namespace customPinMarker {
    const lat_1: number;
    export { lat_1 as lat };
    const lng_1: number;
    export { lng_1 as lng };
    export const type: string;
  }
  namespace customPinMarkerCenter {
    const lat_2: number;
    export { lat_2 as lat };
    const lng_2: number;
    export { lng_2 as lng };
  }
  namespace copy {
    namespace storeLocator {
      const label: string;
      const message: string;
      const title: string;
      const url: string;
    }
    const directions: string;
    namespace openingHours {
      const alternateHoursNote: string;
      const heading: string;
    }
  }
  namespace customPlaceMarker {
    export const id: string;
    const lat_3: number;
    export { lat_3 as lat };
    const lng_3: number;
    export { lng_3 as lng };
    export const storeName: string;
    export const storeType: string;
    const type_1: string;
    export { type_1 as type };
    export const address: string;
    export const phoneNumber: string;
    const openingHours_1: (
      | {
          dayName: string;
          hours: string;
          id: string;
          isAlternate?: undefined;
        }
      | {
          dayName: string;
          hours: string;
          id: string;
          isAlternate: boolean;
        }
    )[];
    export { openingHours_1 as openingHours };
  }
  namespace customPlaceMarkerCenter {
    const lat_4: number;
    export { lat_4 as lat };
    const lng_4: number;
    export { lng_4 as lng };
  }
  const places: (
    | {
        id: string;
        lat: number;
        lng: number;
        storeName: string;
        storeType: string;
        address: string;
        phoneNumber: string;
        openingHours: (
          | {
              dayName: string;
              hours: string;
              id: string;
              isAlternate?: undefined;
            }
          | {
              dayName: string;
              hours: string;
              id: string;
              isAlternate: boolean;
            }
        )[];
      }
    | {
        id: string;
        lat: number;
        lng: number;
        storeName: string;
        storeType: string;
        openingHours: (
          | {
              dayName: string;
              hours: string;
              id: string;
              isAlternate?: undefined;
            }
          | {
              dayName: string;
              hours: string;
              id: string;
              isAlternate: boolean;
            }
        )[];
        address?: undefined;
        phoneNumber?: undefined;
      }
  )[];
}
