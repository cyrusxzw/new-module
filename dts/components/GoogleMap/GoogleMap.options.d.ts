export namespace GoogleMapOptions {
  const SEARCH_TYPES: string[];
  const SEARCH_TYPE_ADDRESS: string;
  const SEARCH_TYPE_GEOCODE: string;
  const SEARCH_TYPE_REGIONS: string;
  const SCROLL_WHEEL_ENABLED: boolean;
  const SHOW_MAP_TYPE_CONTROL: boolean;
  const MAP_INITIAL_ZOOM: number;
  const MAP_MAX_ZOOM: number;
  const MAP_MARKER_ANCHOR_X: number;
  const MAP_MARKER_ANCHOR_Y: number;
  const MAP_LABEL_ORIGIN_X: number;
  const MAP_LABEL_ORIGIN_Y: number;
  namespace MAP_MARKER {
    const path: string;
    const fillColor: string;
    const fillOpacity: number;
    const scale: number;
    const strokeColor: string;
    const strokeWeight: number;
  }
  namespace STOCKIST_MAP_MARKER {
    const path_1: string;
    export { path_1 as path };
    const fillColor_1: string;
    export { fillColor_1 as fillColor };
    const fillOpacity_1: number;
    export { fillOpacity_1 as fillOpacity };
    const scale_1: number;
    export { scale_1 as scale };
    const strokeColor_1: string;
    export { strokeColor_1 as strokeColor };
    const strokeWeight_1: number;
    export { strokeWeight_1 as strokeWeight };
  }
  namespace MAP_MARKER_ACTIVE {
    const path_2: string;
    export { path_2 as path };
    const fillColor_2: string;
    export { fillColor_2 as fillColor };
    const fillOpacity_2: number;
    export { fillOpacity_2 as fillOpacity };
    const scale_2: number;
    export { scale_2 as scale };
    const strokeColor_2: string;
    export { strokeColor_2 as strokeColor };
    const strokeWeight_2: number;
    export { strokeWeight_2 as strokeWeight };
  }
  namespace MAP_MARKER_CLUSTER {
    const height: number;
    const textColor: string;
    const textSize: number;
    const width: number;
  }
  namespace MAP_LABEL {
    const color: string;
    const fontFamily: string;
    const fontSize: string;
    const fontWeight: string;
  }
  namespace MAP_LABEL_ACTIVE {
    const color_1: string;
    export { color_1 as color };
    const fontFamily_1: string;
    export { fontFamily_1 as fontFamily };
    const fontSize_1: string;
    export { fontSize_1 as fontSize };
    const fontWeight_1: string;
    export { fontWeight_1 as fontWeight };
  }
  const MAP_STYLES: (
    | {
        featureType: string;
        elementType: string;
        stylers: (
          | {
              saturation: string;
              color?: undefined;
              lightness?: undefined;
            }
          | {
              color: string;
              saturation?: undefined;
              lightness?: undefined;
            }
          | {
              lightness: string;
              saturation?: undefined;
              color?: undefined;
            }
        )[];
      }
    | {
        featureType: string;
        elementType: string;
        stylers: (
          | {
              visibility: string;
              color?: undefined;
              lightness?: undefined;
              gamma?: undefined;
            }
          | {
              color: string;
              visibility?: undefined;
              lightness?: undefined;
              gamma?: undefined;
            }
          | {
              lightness: string;
              visibility?: undefined;
              color?: undefined;
              gamma?: undefined;
            }
          | {
              gamma: string;
              visibility?: undefined;
              color?: undefined;
              lightness?: undefined;
            }
        )[];
      }
    | {
        featureType: string;
        elementType: string;
        stylers: (
          | {
              lightness: number;
              color?: undefined;
            }
          | {
              color: string;
              lightness?: undefined;
            }
        )[];
      }
    | {
        featureType: string;
        elementType: string;
        stylers: (
          | {
              lightness: number;
              weight?: undefined;
              visibility?: undefined;
            }
          | {
              weight: number;
              lightness?: undefined;
              visibility?: undefined;
            }
          | {
              visibility: string;
              lightness?: undefined;
              weight?: undefined;
            }
        )[];
      }
  )[];
  namespace INFO_BUBBLE_STYLES {
    const arrowSize: number;
    const shadowStyle: number;
    const backgroundColor: string;
    const disableAnimation: boolean;
    const hideCloseButton: boolean;
    const padding: number;
    const borderWidth: number;
    const borderRadius: number;
  }
}
