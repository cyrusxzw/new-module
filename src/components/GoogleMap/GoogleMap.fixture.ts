const GoogleMapFixture = {
  initialZoom: 14,
  center: { lat: -37.8064, lng: 144.9826 },
  customPinMarker: { lat: -37.78863, lng: 144.9847727, type: 'PIN' },
  customPinMarkerCenter: { lat: -37.78863, lng: 144.9847727 },
  copy: {
    storeLocator: {
      label: 'Store locator',
      message: 'Visit our nearby stores.',
      title: 'Open store locator link',
      url: '/',
    },
    directions: 'Directions to',
    openingHours: {
      alternateHoursNote: 'Special opening hours',
      heading: 'Opening hours',
    },
  },
  customPlaceMarker: {
    id: 'aesop-collins-street',
    lat: -37.813971,
    lng: 144.970604,
    storeName: 'Aesop Collins Street',
    storeType: 'SIGNATURE_STORE',
    type: 'PLACE',
    address: '87 Collins Street, Melbourne 3000',
    phoneNumber: '+61 3 9650 3027',
    openingHours: [
      { dayName: 'Monday', hours: '11am - 4pm', id: '1' },
      { dayName: 'Tuesday', hours: '10am - 3pm', id: '2', isAlternate: true },
      { dayName: 'Wednesday', hours: '10am - 3pm', id: '3' },
      { dayName: 'Thursday', hours: '10am - 3pm', id: '4' },
      { dayName: 'Friday', hours: '10am - 3pm', id: '5' },
      { dayName: 'Saturday', hours: '10am - 3pm', id: '6', isAlternate: true },
      { dayName: 'Sunday', hours: '10am - 3pm', id: '7' },
    ],
  },
  customPlaceMarkerCenter: {
    lat: -37.813971,
    lng: 144.970604,
  },
  places: [
    {
      id: 'aesop-fitzroy',
      lat: -37.8064,
      lng: 144.9826,
      storeName: 'Aesop Fitzroy',
      storeType: 'SIGNATURE_STORE',
      address: '242 Gertrude Street, Fitzroy VIC 3065',
      phoneNumber: '+61 3 9419 8356',
      openingHours: [
        { dayName: 'Monday', hours: '11am - 4pm', id: '1' },
        { dayName: 'Tuesday', hours: '10am - 3pm', id: '2', isAlternate: true },
        { dayName: 'Wednesday', hours: '10am - 3pm', id: '3' },
        { dayName: 'Thursday', hours: '10am - 3pm', id: '4' },
        { dayName: 'Friday', hours: '10am - 3pm', id: '5' },
        {
          dayName: 'Saturday',
          hours: '10am - 3pm',
          id: '6',
          isAlternate: true,
        },
        { dayName: 'Sunday', hours: '10am - 3pm', id: '7' },
      ],
    },
    {
      id: 'aesop-qv',
      lat: -37.810711,
      lng: 144.965485,
      storeName: 'Aesop QV',
      storeType: 'SIGNATURE_STORE',
      openingHours: [
        { dayName: 'Monday', hours: '11am - 4pm', id: '1' },
        { dayName: 'Tuesday', hours: '10am - 3pm', id: '2', isAlternate: true },
        { dayName: 'Wednesday', hours: '10am - 3pm', id: '3' },
        { dayName: 'Thursday', hours: '10am - 3pm', id: '4' },
        { dayName: 'Friday', hours: '10am - 3pm', id: '5' },
        {
          dayName: 'Saturday',
          hours: '10am - 3pm',
          id: '6',
          isAlternate: true,
        },
        { dayName: 'Sunday', hours: '10am - 3pm', id: '7' },
      ],
    },
    {
      id: 'aesop-south-yarra',
      lat: -37.8388125,
      lng: 144.9914976,
      storeName: 'Aesop South Yarra',
      storeType: 'SIGNATURE_STORE',
      openingHours: [
        { dayName: 'Monday', hours: '11am - 4pm', id: '1' },
        { dayName: 'Tuesday', hours: '10am - 3pm', id: '2', isAlternate: true },
        { dayName: 'Wednesday', hours: '10am - 3pm', id: '3' },
        { dayName: 'Thursday', hours: '10am - 3pm', id: '4' },
        { dayName: 'Friday', hours: '10am - 3pm', id: '5' },
        {
          dayName: 'Saturday',
          hours: '10am - 3pm',
          id: '6',
          isAlternate: true,
        },
        { dayName: 'Sunday', hours: '10am - 3pm', id: '7' },
      ],
    },
    {
      id: 'myer-melbourne',
      lat: -37.813791,
      lng: 144.964008,
      storeName: 'Myer Melbourne',
      storeType: 'DEPARTMENT_STORE',
      openingHours: [
        { dayName: 'Monday', hours: '11am - 4pm', id: '1' },
        { dayName: 'Tuesday', hours: '10am - 3pm', id: '2', isAlternate: true },
        { dayName: 'Wednesday', hours: '10am - 3pm', id: '3' },
        { dayName: 'Thursday', hours: '10am - 3pm', id: '4' },
        { dayName: 'Friday', hours: '10am - 3pm', id: '5' },
        {
          dayName: 'Saturday',
          hours: '10am - 3pm',
          id: '6',
          isAlternate: true,
        },
        { dayName: 'Sunday', hours: '10am - 3pm', id: '7' },
      ],
    },
    {
      id: 'aesop-newtown',
      lat: -33.893405,
      lng: 151.18367,
      storeName: 'Aesop Newtown',
      storeType: 'SIGNATURE_STORE',
      address: '134A King Street, Newtown, NSW 2042',
      phoneNumber: '+61 2 9517 3319',
      openingHours: [
        { dayName: 'Monday', hours: '11am - 4pm', id: '1' },
        { dayName: 'Tuesday', hours: '10am - 3pm', id: '2', isAlternate: true },
        { dayName: 'Wednesday', hours: '10am - 3pm', id: '3' },
        { dayName: 'Thursday', hours: '10am - 3pm', id: '4' },
        { dayName: 'Friday', hours: '10am - 3pm', id: '5' },
        {
          dayName: 'Saturday',
          hours: '10am - 3pm',
          id: '6',
          isAlternate: true,
        },
        { dayName: 'Sunday', hours: '10am - 3pm', id: '7' },
      ],
    },
  ],
  theme: 'dark',
};

export { GoogleMapFixture };
