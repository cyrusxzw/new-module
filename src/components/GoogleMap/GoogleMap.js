import React, { useCallback, useEffect, useRef, useState } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import isFunction from 'lodash/isFunction';
import { renderToStaticMarkup } from 'react-dom/server.browser';
import MarkerClusterer from '@google/markerclusterer';
import { HYPERLINK_STYLE_TYPES, GOOGLE_MAPS, STORES } from '~/constants';
import { useGoogleMapsContext } from '~/contexts';
import { useWindowHasResized } from '~/customHooks';
import { isViewport } from '~/utils/viewport';
import { Hyperlink } from '~/components/Hyperlink';
import { Loading } from '~/components/Loading';
import { Transition } from '~/components/Transition';
import { GoogleMapOptions } from './GoogleMap.options';
import { InfoCard } from './components/InfoCard/index.ts';
import styles from './GoogleMap.module.css';

const GoogleMap = ({
  center,
  className,
  copy,
  customMarker,
  hasMarkerIndexes,
  id,
  initialZoom,
  places,
}) => {
  const { googleMap, isLoading } = useGoogleMapsContext();
  const mapContainerRef = useRef();
  const mapRef = useRef();
  const activeInfoCard = useRef(null);
  const handleMapClick = useRef(null);
  const isXSmallOrSmallOnlyViewport = useRef(isViewport('xs to sm only'));
  const isMediumViewport = useRef(isViewport('md'));
  const [activeInfoBlockData, setActiveInfoBlockData] = useState(null);
  const [markers, setMarkers] = useState([]);
  const [markerCluster, setMarkerCluster] = useState([]);

  useWindowHasResized(() => {
    if (activeInfoCard.current) {
      activeInfoCard.current.close();
    }

    setActiveInfoBlockData(null);
  });

  const createGoogleMap = useCallback(() => {
    if (googleMap) {
      return new googleMap.maps.Map(mapContainerRef.current, {
        center,
        mapTypeControl: GoogleMapOptions.SHOW_MAP_TYPE_CONTROL,
        scrollwheel: GoogleMapOptions.SCROLL_WHEEL_ENABLED,
        styles: GoogleMapOptions.MAP_STYLES,
        zoom: initialZoom,
        clickableIcons: false,
      });
    }
  }, [center, initialZoom, googleMap]);

  isXSmallOrSmallOnlyViewport.current = isViewport('xs to sm only');
  isMediumViewport.current = isViewport('md');

  const clearMapMarkers = () => {
    markers.forEach(marker => marker.setMap(null));
    setMarkers([]);
  };

  const clearMapClusters = () => {
    if (isFunction(markerCluster.clearMarkers)) {
      markerCluster.clearMarkers();
    }

    setMarkerCluster(null);
  };

  useEffect(() => {
    mapRef.current = createGoogleMap();

    if (googleMap) {
      handleMapClick.current = googleMap.maps.event.addListener(
        mapRef.current,
        'click',
        () => {
          if (activeInfoCard.current) {
            activeInfoCard.current.close();
          }
        },
      );

      setMarkers(() =>
        [customMarker, ...places]
          .filter(item => item?.lat !== undefined && item?.lng !== undefined)
          .map((marker, index) =>
            marker.type === GOOGLE_MAPS.MARKER_TYPE.PIN
              ? createPinMarker(marker, index)
              : createPlaceMarker(marker, index, index === 0),
          ),
      );
    }

    return function cleanup() {
      if (googleMap) {
        googleMap.maps.event.removeListener(handleMapClick.current);
        clearMapMarkers();
      }
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [createGoogleMap, customMarker, places, googleMap]);

  useEffect(() => {
    const options = {
      styles: [
        {
          url: GOOGLE_MAPS.CLUSTER_IMAGE_PATH, // https://www.aesop.com/_ui/responsive/common/images/icons/map-cluster-icon.svg
          ...GoogleMapOptions.MAP_MARKER_CLUSTER,
        },
      ],
    };

    if (googleMap) {
      setMarkerCluster(
        () => new MarkerClusterer(mapRef.current, markers, options),
      );
    }
    return function cleanup() {
      clearMapClusters();
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [markers]);

  const createPinMarker = useCallback(
    (pin, index) => {
      return new googleMap.maps.Marker({
        position: new googleMap.maps.LatLng(
          parseFloat(pin.lat),
          parseFloat(pin.lng),
        ),
        clickable: false,
        map: mapRef.current,
        icon: {
          anchor: new googleMap.maps.Point(
            GoogleMapOptions.MAP_MARKER_ANCHOR_X,
            GoogleMapOptions.MAP_MARKER_ANCHOR_Y,
          ),
          labelOrigin: new googleMap.maps.Point(
            GoogleMapOptions.MAP_LABEL_ORIGIN_X,
            GoogleMapOptions.MAP_LABEL_ORIGIN_Y,
          ),
          ...GoogleMapOptions.MAP_MARKER,
        },
        label: {
          text: hasMarkerIndexes ? `${index + 1}` : ' ',
          ...GoogleMapOptions.MAP_LABEL,
        },
      });
    },
    [hasMarkerIndexes, googleMap],
  );

  const createPlaceMarker = useCallback(
    (
      { address, lat, lng, openingHours, phoneNumber, storeName, storeType },
      index = 0,
      custom = false,
    ) => {
      const isStockistMarker =
        storeType === STORES.LOCATION_TYPES.STOCKIST ||
        storeType === STORES.LOCATION_TYPES.DEPARTMENT_STORE;

      const markerType =
        !custom && (customMarker || isStockistMarker)
          ? GoogleMapOptions.STOCKIST_MAP_MARKER
          : GoogleMapOptions.MAP_MARKER;

      const marker = new googleMap.maps.Marker({
        position: new googleMap.maps.LatLng(parseFloat(lat), parseFloat(lng)),
        map: mapRef.current,
        icon: {
          anchor: new googleMap.maps.Point(
            GoogleMapOptions.MAP_MARKER_ANCHOR_X,
            GoogleMapOptions.MAP_MARKER_ANCHOR_Y,
          ),
          labelOrigin: new googleMap.maps.Point(
            GoogleMapOptions.MAP_LABEL_ORIGIN_X,
            GoogleMapOptions.MAP_LABEL_ORIGIN_Y,
          ),
          ...markerType,
        },
        title: storeName,
        label: {
          text: hasMarkerIndexes ? `${index + 1}` : ' ',
          ...GoogleMapOptions.MAP_LABEL,
        },
      });

      const infoCard = new googleMap.maps.InfoWindow({
        content: renderToStaticMarkup(
          <InfoCard
            address={address}
            copy={{
              directions: copy.directions,
              openingHours: copy.openingHours,
            }}
            count={index + 1}
            openingHours={openingHours}
            phoneNumber={phoneNumber}
            storeName={storeName}
          />,
        ),
      });

      marker.addListener('click', () => {
        if (isMediumViewport.current) {
          if (activeInfoCard.current) {
            activeInfoCard.current.close();
          }

          activeInfoCard.current = infoCard;
          activeInfoCard.current.open(mapRef.current, marker);
        }

        if (isXSmallOrSmallOnlyViewport.current) {
          setActiveInfoBlockData({
            address,
            count: index + 1,
            phoneNumber,
            storeName,
            openingHours,
          });
        }
      });

      return marker;
    },
    [
      googleMap,
      copy,
      customMarker,
      hasMarkerIndexes,
      isMediumViewport,
      isXSmallOrSmallOnlyViewport,
    ],
  );

  const classSet = cx(styles.base, className);

  return (
    <div className={classSet}>
      <div className={styles.wrapper}>
        <Loading className={styles.loading} isLoading={isLoading} />
        <div className={styles.map} id={id} ref={mapContainerRef} />
      </div>
      <Transition
        isActive={!!activeInfoBlockData}
        shouldMountOnEnter={true}
        shouldUnmountOnExit={true}
        type="shiftInDown"
      >
        <InfoCard
          address={activeInfoBlockData?.address}
          className={styles.infoCardBlock}
          copy={{
            directions: copy.directions,
            openingHours: copy.openingHours,
          }}
          count={activeInfoBlockData?.count}
          hasMarkerIndexes={hasMarkerIndexes}
          openingHours={activeInfoBlockData?.openingHours}
          phoneNumber={activeInfoBlockData?.phoneNumber}
          storeName={activeInfoBlockData?.storeName}
        />
      </Transition>
      <footer className={styles.footer}>
        <div className={styles.viewStoreLabel}>
          {copy.storeLocator?.message}
        </div>
        <div className={styles.viewStoreLinkWrapper}>
          <Hyperlink
            className={styles.viewStoreLink}
            style={HYPERLINK_STYLE_TYPES.INTERNAL_TEXT_LINK}
            title={copy.storeLocator?.title}
            url={copy.storeLocator?.url}
          >
            {copy.storeLocator?.label}
          </Hyperlink>
        </div>
      </footer>
    </div>
  );
};

GoogleMap.propTypes = {
  center: PropTypes.shape({
    lat: PropTypes.number,
    lng: PropTypes.number,
  }).isRequired,
  className: PropTypes.string,
  copy: PropTypes.shape({
    directions: PropTypes.string,
    storeLocator: PropTypes.shape({
      label: PropTypes.string,
      message: PropTypes.string,
      title: PropTypes.string,
      url: PropTypes.string,
    }),
    openingHours: PropTypes.shape({
      alternateHoursNote: PropTypes.string,
      heading: PropTypes.string,
    }),
  }),
  customMarker: PropTypes.shape({
    lat: PropTypes.number,
    lng: PropTypes.number,
    type: PropTypes.oneOf(['PIN', 'PLACE']),
  }),
  hasMarkerIndexes: PropTypes.bool,
  id: PropTypes.string,
  initialZoom: PropTypes.number,
  places: PropTypes.arrayOf(
    PropTypes.shape({
      lat: PropTypes.number,
      lng: PropTypes.number,
      id: PropTypes.string,
      storeName: PropTypes.string,
      storeType: PropTypes.string,
      address: PropTypes.string,
      phoneNumber: PropTypes.string,
      openingHours: PropTypes.array,
    }),
  ),
};

GoogleMap.defaultProps = {
  center: {},
  className: undefined,
  copy: {
    directions: 'Directions to',
    storeLocator: {
      label: 'Store locator',
      message: 'Visit our nearby stores.',
      title: 'Open store locator link',
      url: '/',
    },
    openingHours: {
      alternateHoursNote: 'Special opening hours',
      heading: 'Opening hours',
    },
  },
  customMarker: undefined,
  hasMarkerIndexes: true,
  id: undefined,
  initialZoom: GoogleMapOptions.MAP_INITIAL_ZOOM,
  places: [],
};

export { GoogleMap };
