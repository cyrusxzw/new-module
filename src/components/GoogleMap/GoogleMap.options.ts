const GoogleMapOptions = {
  SEARCH_TYPES: ['address', 'geocode', '(regions)'],
  SEARCH_TYPE_ADDRESS: 'address',
  SEARCH_TYPE_GEOCODE: 'geocode',
  SEARCH_TYPE_REGIONS: '(regions)',
  SCROLL_WHEEL_ENABLED: false,
  SHOW_MAP_TYPE_CONTROL: false,
  MAP_INITIAL_ZOOM: 13,
  MAP_MAX_ZOOM: 15,
  MAP_MARKER_ANCHOR_X: 14,
  MAP_MARKER_ANCHOR_Y: 14,
  MAP_LABEL_ORIGIN_X: 14,
  MAP_LABEL_ORIGIN_Y: 14,
  MAP_MARKER: {
    path:
      'M14.501,1.500 C21.681,1.500 27.500,7.320 27.500,14.500 C27.500,21.679 21.681,27.500 14.501,27.500 C7.320,27.500 1.500,21.679 1.500,14.500 C1.500,7.320 7.320,1.500 14.501,1.500 Z',
    fillColor: '#333333',
    fillOpacity: 1,
    scale: 1,
    strokeColor: '#333333',
    strokeWeight: 1,
  },
  STOCKIST_MAP_MARKER: {
    path:
      'M14.501,1.500 C21.681,1.500 27.500,7.320 27.500,14.500 C27.500,21.679 21.681,27.500 14.501,27.500 C7.320,27.500 1.500,21.679 1.500,14.500 C1.500,7.320 7.320,1.500 14.501,1.500 Z',
    fillColor: '#848482',
    fillOpacity: 1,
    scale: 1,
    strokeColor: '#848482',
    strokeWeight: 1,
  },
  MAP_MARKER_ACTIVE: {
    path:
      'M14.501,1.500 C21.681,1.500 27.500,7.320 27.500,14.500 C27.500,21.679 21.681,27.500 14.501,27.500 C7.320,27.500 1.500,21.679 1.500,14.500 C1.500,7.320 7.320,1.500 14.501,1.500 Z',
    fillColor: '#fefef2',
    fillOpacity: 1,
    scale: 1,
    strokeColor: '#333333',
    strokeWeight: 1,
  },
  MAP_MARKER_CLUSTER: {
    height: 34,
    textColor: 'rgba(0, 0, 0, 0)',
    textSize: 0,
    width: 34,
  },
  MAP_LABEL: {
    color: '#fefef2',
    fontFamily: 'Suisse Medium, sans-serif',
    fontSize: '11px',
    fontWeight: 'bold',
  },
  MAP_LABEL_ACTIVE: {
    color: '#333333',
    fontFamily: 'Suisse Medium, sans-serif',
    fontSize: '11px',
    fontWeight: 'bold',
  },
  MAP_STYLES: [
    {
      featureType: 'all',
      elementType: 'labels.text.fill',
      stylers: [
        {
          saturation: '0',
        },
        {
          color: '#a1a195',
        },
        {
          lightness: '0',
        },
      ],
    },
    {
      featureType: 'all',
      elementType: 'labels.text.stroke',
      stylers: [
        {
          visibility: 'on',
        },
        {
          color: '#fefef8',
        },
        {
          lightness: '0',
        },
        {
          gamma: '1',
        },
      ],
    },
    {
      featureType: 'all',
      elementType: 'labels.icon',
      stylers: [
        {
          visibility: 'off',
        },
      ],
    },
    {
      featureType: 'administrative',
      elementType: 'geometry.fill',
      stylers: [
        {
          lightness: 20,
        },
        {
          color: '#ebebdf',
        },
      ],
    },
    {
      featureType: 'administrative',
      elementType: 'geometry.stroke',
      stylers: [
        {
          lightness: 17,
        },
        {
          weight: 1.2,
        },
      ],
    },
    {
      featureType: 'administrative.country',
      elementType: 'labels.text.fill',
      stylers: [
        {
          color: '#808077',
        },
        {
          saturation: '0',
        },
      ],
    },
    {
      featureType: 'administrative.province',
      elementType: 'labels.text.fill',
      stylers: [
        {
          color: '#808077',
        },
      ],
    },
    {
      featureType: 'administrative.locality',
      elementType: 'labels.text.fill',
      stylers: [
        {
          color: '#808077',
        },
      ],
    },
    {
      featureType: 'landscape',
      elementType: 'geometry',
      stylers: [
        {
          color: '#f4f4e8',
        },
        {
          lightness: '0',
        },
        {
          gamma: '1',
        },
      ],
    },
    {
      featureType: 'poi',
      elementType: 'geometry',
      stylers: [
        {
          color: '#ebebdf',
        },
        {
          lightness: '0',
        },
        {
          gamma: '1.00',
        },
      ],
    },
    {
      featureType: 'road',
      elementType: 'geometry.fill',
      stylers: [
        {
          color: '#fefef8',
        },
      ],
    },
    {
      featureType: 'road.highway',
      elementType: 'geometry.stroke',
      stylers: [
        {
          lightness: 29,
        },
        {
          weight: 0.2,
        },
        {
          visibility: 'off',
        },
      ],
    },
    {
      featureType: 'transit',
      elementType: 'geometry',
      stylers: [
        {
          color: '#ebebdf',
        },
      ],
    },
    {
      featureType: 'water',
      elementType: 'geometry',
      stylers: [
        {
          color: '#deded3',
        },
        {
          lightness: 17,
        },
      ],
    },
  ],
  INFO_BUBBLE_STYLES: {
    arrowSize: 10,
    shadowStyle: 0,
    backgroundColor: '#FEFFF8',
    disableAnimation: true,
    hideCloseButton: true,
    padding: 0,
    borderWidth: 0,
    borderRadius: 0,
  },
};

export { GoogleMapOptions };
