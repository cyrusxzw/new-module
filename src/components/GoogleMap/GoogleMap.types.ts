import type { ComponentWithoutChildren, Themes } from '~/types';

type GoogleMapProps = {
  center: {
    lat?: number;
    lng?: number;
  };
  className?: string;
  copy?: {
    directions?: string;
    storeLocator?: {
      label?: string;
      message?: string;
      title?: string;
      url?: string;
    };
    openingHours?: {
      alternateHoursNote?: string;
      heading?: string;
    };
  };
  customMarker?: {
    lat?: number;
    lng?: number;
    type?: 'PIN' | 'PLACE';
  };
  hasMarkerIndexes?: boolean;
  id?: string;
  initialZoom?: number;
  places?: {
    lat?: number;
    lng?: number;
    id?: string;
    storeName?: string;
    storeType?: string;
    address?: string;
    phoneNumber?: string;
    openingHours?: any[];
  }[];
  theme?: Themes;
};

type GoogleMapType = ComponentWithoutChildren<GoogleMapProps>;

export type { GoogleMapType, GoogleMapProps };
