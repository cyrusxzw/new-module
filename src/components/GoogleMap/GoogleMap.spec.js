import React from 'react';
import { render } from '@testing-library/react';
import { GoogleMapFixture } from './GoogleMap.fixture';
import { GoogleMap } from './GoogleMap';

describe('<GoogleMap />', () => {
  it('should render base component correctly', () => {
    const { container } = render(
      <GoogleMap
        center={GoogleMapFixture.customPinMarkerCenter}
        copy={GoogleMapFixture.copy}
        customMarker={GoogleMapFixture.customPinMarker}
        hasMarkerIndexes={false}
        id="google-map"
        initialZoom={GoogleMapFixture.initialZoom}
        places={GoogleMapFixture.places}
        theme={GoogleMapFixture.theme}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
