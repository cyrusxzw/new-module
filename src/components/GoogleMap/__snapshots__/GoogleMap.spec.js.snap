// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoogleMap /> should render base component correctly 1`] = `
<div>
  <div
    class="base"
  >
    <div
      class="wrapper"
    >
      <span
        class="base center medium dark loading"
        data-testid="data-testid-loading"
        role="status"
      />
      <div
        class="map"
        id="google-map"
      />
    </div>
    <footer
      class="footer"
    >
      <div
        class="viewStoreLabel"
      >
        Visit our nearby stores.
      </div>
      <div
        class="viewStoreLinkWrapper"
      >
        <a
          class="base hasIcon left dark viewStoreLink"
          href="/"
          target="_self"
          title="Open store locator link"
        >
          Store locator
           
          <i
            aria-hidden="true"
            class="icon"
          >
            <svg
              aria-labelledby="rightArrow-1"
              class="base rightArrow dark"
              data-testid="data-testid-Icon"
              focusable="false"
              height="12"
              role="img"
              style="height: 12px; width: 12px;"
              tabindex="-1"
              viewBox="0 0 50 50"
              width="12"
            >
              <title
                id="rightArrow-1"
              >
                rightArrow
              </title>
              <g>
                <path
                  d="M30.1,5.3L50,25.1L30.1,45h-6.6l18-17.6H0v-4.8h41.5l-18-17.6h6.6V5.3z"
                />
              </g>
            </svg>
          </i>
        </a>
      </div>
    </footer>
  </div>
</div>
`;
